Code Clean Up

Platnumrcn14 opened the pull request:
**Resolves issue #15**
-Placed each class into its own file within components folder
-Renamed variables to match python schema where possible
-removed init.py from components folder


![Screenshot 2025-02-23 042318](https://github.com/user-attachments/assets/0e7d3b97-2f71-4c1d-948c-10e729d7cf58)

Platnumrcn14 committed:
-modified variable names where feasable to match python conventions

Platnumrcn14 committed:
-compartmentalized each class of components/__init__.py and taskchampion.py within the components folder.
-added nescessary import statements for each new class file created

Platnumrcn14 committed:
-updated to match latest version of develop
-Moved ALIGN class into independent folder
-Moved COLS constnat into task_row.py file
-made some further python schema corrections

Platnumrcn14 committed:
-cleaned up import statements to properly link each component

Platnumrcn14 committed:
-further import statement code cleanup

Platnumrcn14 committed:
-removed init.py from components

modmo22 reviewed (dismissed):
You should change the last modified dates on all of these modules that you've touched and rename TableCell to either tablecell or table_cell so that when that class is imported it is distinct from the module. 

So, basically the way it's written now, to import the class, it would look like `from TableCell import TableCell`. If you rename the module, the import statement would be the more intentional `from table_cell import TableCell`.

jacobwilki01 commented on components/add_task_dialog.py:
```diff
@@ -0,0 +1,97 @@
+    def add_task(self) -> Optional[TaskDetails]:
+        if self.exec():
+            if self.is_recurring:
+                test = self.due_date.dateTime().toPython()
```
delete this. this might have been something I left

modmo22 replied:
Actually, if you're going to remove the snakecase, then we'd want the class files to still be undercase for the reason that I cited in my original comment. The suggestion to capitalize the class names violates PEP8 style guidelines.

jacobwilki01 commented on taskchampion.py:
```diff
@@ -20,175 +20,21 @@
-        app.mainWidget.grids[0].addTask(Task(task))  # Add the task to the first grid.
-    
-    app.loadStyles()  # Load the styles.
+    # # Add both pending and completed tasks.
```
remove this commented out code.

jacobwilki01 reviewed (changes requested):
Few things:
---Rename all of the class files to mirror the actual class name. e.g., AddTaskDialog instead of add_task_dialog.
---Run through your IDE and see if every set of imports throughout the project are still necessary. My IDE is showing that a lot of them are irrelevant.
---Update all prologue comments to include your name + the correct date modified.
---Organize the components folder a little bit more, instead of just having all of the class files in there. Example: split up "GUI" elements from "Dialogs".
---Split out "__init__.py" into API and Logger classes as well? Then, store the instantiation of each singleton in that individual class file?

Platnumrcn14 committed:
-updated folder naming
-updated dates and names on modules
-organized components folder into 'GUI' and 'Dialogs' folders
-some breakup of utils/init.py completed

Platnumrcn14 committed:
-broke up utils/init.py into 4 additional class files in utils
-updated prologue comments
-added Dialogs and GUI folders to further compartmentalize component folder contents
-removed irrelevant import statements
-updated code to match develop before implementing changes

Platnumrcn14 committed:
-corrected indenting error
-removed excess task_row.py file in components

jacobwilki01 commented on components/GUI/buttonbox.py:
```diff
@@ -1,22 +1,36 @@
+from .tablecell import TableCell
+from utils.task import Task
 
 class Buttonbox(TableCell):
```
Rename to `ButtonBox`.

Platnumrcn14 committed:
-Renamed 'Buttonbox' to 'ButtonBox'

Platnumrcn14 committed:
-fixed merge conflicts
-ammended checkbox.py to match develop branch
